lib/openscap_parser/value_definition.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class ValueDefinition < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def description
+      @description ||= newline_to_whitespace(
+        parsed_xml.at_css('description') &&
+          parsed_xml.at_css('description').text || ''
+      )
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title') &&
+        parsed_xml.at_css('title').text
+    end
+
+    def type
+      @type ||= parsed_xml['type'] || 'string'
+    end
+
+    def lower_bound
+      lower_bound_element = parsed_xml.at_xpath("lower-bound[@selector='']") || parsed_xml.at_xpath('lower-bound[not(@selector)]')
+      @lower_bound ||= lower_bound_element&.text
+    end
+
+    def upper_bound
+      upper_bound_element = parsed_xml.at_xpath("upper-bound[@selector='']") || parsed_xml.at_xpath('upper-bound[not(@selector)]')
+      @upper_bound ||= upper_bound_element&.text
+    end
+
+    def default_value
+      # The default value is the value or complex-value element with a empty or absent @selector
+      # If there is no value or complex-value element with an empty or absent @selector, the first value in
+      # the top down processing shall be the default element
+      # There should only be at most 1 value or complex value with an empty or absent @selector
+      value_element = parsed_xml.at_xpath("value[@selector='']") || parsed_xml.at_xpath('value[not(@selector)]') ||
+        parsed_xml.at_xpath("complex-value[@selector='']") || parsed_xml.at_xpath('complex-value[not(@selector)]') ||
+        parsed_xml.xpath("value | complex-value")[0]
+      @default_value ||= value_element&.text
```

Comment:
Yeah, this one is complicated. I can split them up into 2. There aren't any actual examples of complex-value in the XCCDF specs or any of the datastreams but I assumed it would look something like this: `<complex-value>604800, 2592000, 86400</complex-value>`
I need to check with the openscap team though to see exactly what that will look like but I figured we could just treat the complex value as a string because even though it is a "list" I didn't think there was a reason that we'd need to return it as an array. So using the complex-value above, the complex value would be `"604800, 2592000, 86400"` instead of `[604800, 2592000, 86400]`

Reply:
But how does that map into our DB model then? :thinking:

Reply:
I think it'd map to our DB model easily because we can just have a `value` column on the `values` table and the column type can be string so we'd store every value as a string and then store the type in a `value_type` column. I think this would be fine because we're not really doing anything with the value besides updating it sometimes so I think as long as we have the value and know what the type is we should be fine. What do you think?

Reply:
I don't know, I think it should be an array :thinking: it's hard without examples, but logically that would make the most sense.

Reply:
I asked the openscap team what the format of complex-value is but I agree that an array would make the most sense